jdbc/src/main/java/org/jclouds/jdbc/service/JdbcService.java

```diff
@@ -16,54 +16,243 @@
+         BlobToBlobEntity blobToBlobEntity) {
       this.containerRepository = containerRepository;
+      this.blobRepository = blobRepository;
+      this.chunkRepository = chunkRepository;
+      this.blobToBlobEntity = blobToBlobEntity;
    }
 
    @Transactional
-   public void createContainer(String name, ContainerAccess access) {
-      containerRepository.create(Container.builder().name(name).containerAccess(access).build());
+   public void createContainer(String containerName, ContainerAccess access) {
+      containerRepository.create(ContainerEntity.builder().name(containerName).containerAccess(access).build());
    }
 
    @Transactional
-   public void createContainer(String name) {
-      containerRepository.create(Container.builder().name(name).build());
+   public void createContainer(String containerName) {
+      createContainer(containerName, null);
    }
 
    @Transactional
-   public List<Container> findAllContainers() {
+   public List<ContainerEntity> findAllContainers() {
       return containerRepository.findAllContainers();
    }
 
    @Transactional
-   public Container findContainerByName(String name) {
-      return containerRepository.findContainerByName(name);
+   public ContainerEntity findContainerByName(String containerName) {
+      return containerRepository.findContainerByName(containerName);
+   }
+
+   @Transactional
+   public void deleteContainerByName(String containerName) {
+      containerRepository.deleteContainerByName(containerName);
+   }
+
+   @Transactional
+   public void setContainerAccessByName(String containerName, ContainerAccess access) {
+      ContainerEntity containerEntity = containerRepository.findContainerByName(containerName);
+      containerEntity.setContainerAccess(access);
+      containerRepository.save(containerEntity);
+   }
+
+   @Transactional
+   public boolean blobExists(String containerName, String key) {
+      return findBlobById(containerName, key) != null;
+   }
+
+   @Transactional(rollbackOn = IOException.class)
+   public BlobEntity createOrModifyBlob(String containerName, Blob blob, BlobAccess blobAccess) throws IOException {
+      String key = blob.getMetadata().getName();
+      Date creationDate = null;
+      BlobEntity oldBlobEntity = findBlobById(containerName, key);
+      if (oldBlobEntity != null) {
+         creationDate = oldBlobEntity.getCreationDate();
+         deleteBlob(containerName, key);
+      }
+      BlobEntity blobEntity = blobToBlobEntity.apply(blob);
+      blobEntity.getPayload().setChunks(storeData(blob.getPayload().openStream()));
+      blobEntity.setContainerEntity(containerRepository.findContainerByName(containerName));
+      blobEntity.setKey(key);
+      blobEntity.setBlobAccess(blobAccess);
+      blobEntity.setCreationDate(creationDate);
+
+
+      HashCode hash = ByteStreams2.hashAndClose(blob.getPayload().openStream(), Hashing.md5());
```

Comment:
Somewhat of a small problem with this.
In a lot of cases, the user has already checked the hash of the stream they are trying to upload to a cloud/jdbc/etc. Which means that by having us to it again, performance is reduced overall. That is really the point of returning the hash.
I am mostly OK with keeping it this way, but maybe explicitly document that the check happens here too? Either that or remove the integrity check. Removing it might be better. WDYT?